src/display.jl

```diff
@@ -1,379 +1,405 @@
 
 """
-    @format [style::Symbol] [decorations::Bool] [sigfigs::Integer]
+    @format [format::Symbol] [decorations::Bool] [sigdigits::Integer]
 
-The `@format` macro provides a simple interface to control the output format
-for intervals. These options are passed to the `setformat` function.
-It returns the new `DisplayParameters` object.
+Change the format used by `show` to display intervals. These options are passed
+to the `setformat` function.
 
-The arguments may be in any order and of type:
+Possible options:
+- `format` can be:
+    - `:standard`: `[1, 2]`
+    - `:midpoint`: display `x::Interval` in the form "mid(x) ± radius(x)"
+    - `:full`: display the entire bounds regardless of `sigdigits`.
+- `sigdigits`: number (greater or equal to 1) of significant digits to display.
+- `decorations`: display the decorations or not.
 
-- `Symbol`: the output format (`:full`, `:standard` or `:midpoint`)
-- `Bool`: whether to display decorations
-- `Integer`: the number of significant figures
+See also: `setformat`.
 
-E.g.
+# Example
 ```
-julia> x = 0.1..0.3
-@[0.0999999, 0.300001]
+julia> x = interval(0.1, 0.3)  # Default display options
+[0.0999999, 0.300001]
 
-julia> @format full
+julia> @format full  # Equivalent to `setformat(:full)``
 Display parameters:
-- format: full
-- decorations: false
-- significant figures: 6
+  - format: full
+  - decorations: false
+  - significant digits: 6
 
 julia> x
 Interval(0.09999999999999999, 0.30000000000000004)
 
-julia> @format standard 3
+julia> y = DecoratedInterval(x)
+[0.0999, 0.301]
+
+julia> @format standard true 3  # Equivalent to `setformat(:standard; decorations = true, sigdigits = 3)`
 Display parameters:
-- format: standard
-- decorations: false
-- significant figures: 3
+  - format: standard
+  - decorations: true
+  - significant digits: 3
 
-julia> x
-[0.0999, 0.301]
+julia> y
+[0.0999, 0.301]_com
 ```
 """
 macro format(expr...)
-
     format = Meta.quot(display_params.format)
     decorations = display_params.decorations
-    sigfigs = display_params.sigfigs
-
+    sigdigits = display_params.sigdigits
     for ex in expr
-
         if isa(ex, Symbol)
             format = Meta.quot(ex)
-
         elseif isa(ex, Bool)
             decorations = ex
-
         elseif isa(ex, Integer)
-            sigfigs = ex
+            sigdigits = ex
+        else
+            return :(throw(ArgumentError("Only accept arguments of type `Symbol`, `Bool` or `Int` but received $($ex)")))
         end
-
     end
-
-    format_code = :(setformat($format, decorations=$decorations, sigfigs=$sigfigs))
-
-    return format_code
+    return :(setformat($format; decorations = $decorations, sigdigits = $sigdigits))
 end
 
+# Printing mechanism for various types of intervals
 
-if VERSION < v"1.1.0-DEV.683"
-    to_mpfr(r) = Base.MPFR.to_mpfr(r)
-else
-    to_mpfr(r) = convert(Base.MPFR.MPFRRoundingMode, r)
-end
-
-## Output
+show(io::IO, a::Union{Interval,Complex{<:Interval},DecoratedInterval,IntervalBox}) =
```

Comment:
Should this be general or only define for `MIME"text/plain"` This also overrides `repr` which is [kind of](https://docs.julialang.org/en/v1/base/strings/#Base.repr-Tuple%7BAny%7D) expected to return a string that can be copy pasted back into julia and work.

Then an even more specific method could be defined for the `:compact:` case, when intervals are printed inside an array.

@lucaferranti @lbenet 

(note that this is also not central to the PR, and can be left for another one)

Reply:
Interesting. I just tried this and I think it is a nice change. As an added benefit, this allows us to remove `showfull` entirely since `show(:Interval)` now has this behaviour.